internal/host/validator.go

```diff
@@ -775,8 +775,28 @@ func (v *validator) belongsToL2MajorityGroup(c *validationContext, majorityGroup
 	// TODO(mko) This rule should be revised as soon as OCP supports multiple machineNetwork
 	//           entries using the same IP stack.
 	ret := true
+
+	groupsForNetwork := func(ipnet *net.IPNet) []strfmt.UUID {
+		ones, bits := ipnet.Mask.Size()
+		for key, groups := range majorityGroups {
+			_, groupIpnet, err := net.ParseCIDR(key)
+			if err != nil {
+				continue
+			}
+			groupOnes, groupBits := groupIpnet.Mask.Size()
+			if groupIpnet.IP.Equal(ipnet.IP) && ones == groupOnes && bits == groupBits {
+				return groups
+			}
+		}
+		return nil
+	}
+
 	for _, machineNet := range c.cluster.MachineNetworks {
-		ret = ret && funk.Contains(majorityGroups[string(machineNet.Cidr)], *c.host.ID)
+		_, machineIpnet, err := net.ParseCIDR(string(machineNet.Cidr))
+		if err != nil {
+			return ValidationError
+		}
+		ret = ret && funk.Contains(groupsForNetwork(machineIpnet), *c.host.ID)
```

Comment:
does that mean that all the groups will be part of the network? 
I mean you have true, true, false the reply will be false no? 
Maybe i missed the logic here

Reply:
Tried to enhance the code.  Maybe it will be clearer.